### DISCUSSION TITLE
How to flash Pinecil V2?

### DISCUSSION THREAD

**Discussion by AchillesPDX** [Ralim/IronOS#1518] [Q&A]
Just got my Pinecil V2 and it came with v2.18.A1A569A but it seems like v2.20 has quite a few bugfixes. Is there any kind of instructions on how to update?

Thanks!

**Comment by croaker-1** [comment]
The most straightforward way is to using the `blisp` command line tool. You can find pre-built binaries for Linux and Windows and instructions on the Pine64 wiki [here](https://wiki.pine64.org/wiki/Pinecil#Update_Pinecil_V2).

Currently `blisp` does not run on macOS although preliminary support for macOS has recently been added to the source code. It seemed to build and run fine but it was not able to flash my Pinecil V2. As a workaround, I setup a Linux VM to run `blisp`...not an ideal setup but it did the job with a minimal amount of work.

**Comment by builder555** [comment]
@croaker-1 latest `blisp` runs on mac (tested on intel and m1): https://github.com/pine64/blisp/releases/latest

I just used it to flash Pinecil: `./blisp write -c bl70x --reset Pinecilv2_EN.bin`

**Comment by River-Mochi** [comment]
Pinecil Wiki  has full firmware instructions to flash Pinecil V2 ( and V1) 

- [X] **BLISP** now works for 🪟 Windows, 🐧 Linux, 🍏 MacOS
- [X] Pinecil Wiki [update instructions](https://wiki.pine64.org/wiki/Pinecil_Firmware)  
- [X] can also build BLISP from code if you have different needs; I was able to build from code for Pinebook Pro (ARM) [Build from code instructions here ](https://github.com/pine64/blisp/wiki/Update-Pinecil-V2)
